### PR TITLE
Support for cds.Vector

### PR DESCRIPTION
**Pull request by stewsk**

### CODE REVIEW

**Comment by stewsk** [comment]
@johannes-vogel @agoerler : as discussed, have started this PR for documenting cds.Vector

**Comment by agoerler** [comment]
@MattSchur - please add Java-specific parts also here

File: java/cds-data.md [hunk 1]
```diff
@@ -276,6 +277,58 @@ person.toJson(); // { "salutation" : "Mr.", name : { "first" : "Frank" } }
...
+To compute the similarity and distance of embeddings in the vector space, you can use the `CQL.cosineSimilarity` or `CQL.l2Distance` (Euclidean distance) functions in queries:
+
+```Java
+//  Compute vector embedding, e.g. via LangChain4j
+float[] embedding = llm.embed(text).content().vector();
+
+CqnSelect query = Select.from(BOOKS).where(b -> // [!code focus]
+  CQL.cosineSimilarity(b.embedding(), CQL.vector(embedding)).gt(0.9)) // [!code focus]
+
+Result relatedBooks = service.run(query); // [!code focus]
```
Comment (agoerler) [inline java/cds-data.md:313]: In the typed API we could introduce a 
* `VectorRef extends ElementRef<CdsVector>`
* Let the builder interface return a `VectorRef` for elements of type `cds.Vector`

And then we could 

```java
CqnSelect query = Select.from(BOOKS)
    .where(b -> b.embedding().cosineSimilarity(CdsVector.of(embedding)).gt(0.9));
```

**Review by agoerler — APPROVED** [review]

**Review by agoerler — APPROVED** [review]

File: java/cds-data.md [hunk 2]
```diff
@@ -276,6 +277,61 @@ person.toJson(); // { "salutation" : "Mr.", name : { "first" : "Frank" } }
...
+    .type(CdsBaseType.VECTOR)).gt(0.9)
+
+Result relatedBooks = service.run(query,
+  Map.of("embedding", CdsVector.of(embedding)));
+```
+
+In CDS QL queries, elements of type `cds.Vector` are not included in select _all_; they must be explicitly added to the select list:
+
+```Java
+CdsVector embedding = service.run(Select.from(BOOKS).byId(101)
```
Comment (smahati) [inline java/cds-data.md:330]: The previous statement makes it seem like only select _all_ doesn't include vectors but as per the code it seems like it affects all selects?

**Review by smahati — APPROVED** [review]

File: java/cds-data.md [hunk 3]
```diff
@@ -276,6 +277,61 @@ person.toJson(); // { "salutation" : "Mr.", name : { "first" : "Frank" } }
 Avoid cyclic relationships between CdsData objects when using toJson.
 :::
 
+
+## Vector Embeddings <Badge type="warning" text="beta" /> { #vector-embeddings }
+
+Vector embeddings are numerical representations that capture features and inherent semantics of unstructured data - such as text, images, or audio. They facilitate tasks like similarity search, recommendations, and Retrieval Augmented Generation (RAG). This improves the results of generative AI (GenAI) by augmenting prompts with relevant data retrieved from a vector datastore such as the [SAP HANA Cloud Vector Engine](https://community.sap.com/t5/technology-blogs-by-sap/sap-hana-cloud-s-vector-engine-announcement/ba-p/13577010).
```
Comment (johannes-vogel) [inline java/cds-data.md:283]: Shouldn't this section be a generic section for both runtimes?
We could include it here https://pages.github.tools.sap/cap/docs/guides/databases-hana#native-sap-hana-features and link the implementation details from there.

@agoerler @MattSchur What do you think?

Comment (MattSchur) [reply java/cds-data.md:283]: Until we add support for other DBs we can move it to the HANA section: https://github.com/cap-js/docs/pull/768

**Pull request merged**